### PR TITLE
lmp-device-auto-register: Improving User Guide

### PR DESCRIPTION
**Pull request by munoz0raul**
Signed-off-by: Raul Muñoz <raul@foundries.io>

### CODE REVIEW

**Comment by doanac** [comment]
Docs for e9522d3215424f53e6b471988d5973034ec51319 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1112/docs/artifacts/html/index.html

File: source/user-guide/lmp-device-auto-register/lmp-device-auto-register.rst [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+This section shows how to enable the ``lmp-device-auto-register`` recipe. This 
+recipe creates a systemd oneshot service that will automatically register the 
+device on first boot once it has internet connectivity.
+This is done by providing an API Token that has **devices:create** scope.
+
+.. warning::
+   Do not use this recipe in production. This recipe is only intended for
+   usage in a development environment. For more information, 
+   read :ref:`ref-factory-registration-ref`.
```
Comment (vanmaegima) [inline source/user-guide/lmp-device-auto-register/lmp-device-auto-register.rst:14]: should we add that for production there is a more secure way to achieve the auto registration and that the customers should contact support in this case?

File: source/user-guide/lmp-device-auto-register/lmp-device-auto-register.rst [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+          host:~$ git add recipes-samples/images/lmp-factory-image.bb
+          host:~$ git add recipes-support/lmp-device-auto-register/lmp-device-auto-register/api-token
+          host:~$ git add recipes-support/lmp-device-auto-register/lmp-device-auto-register.bbappend
+          host:~$ git commit -m "lmp-device-auto-register: Adding recipe"
+          host:~$ git push
+
+The latest **Target** named ``platform-devel`` should be the CI job you just created.
+
+When FoundriesFactory CI finishes all jobs, download and flashes the image.
```
Comment (vanmaegima) [inline source/user-guide/lmp-device-auto-register/lmp-device-auto-register.rst:133]:
```suggestion
When FoundriesFactory CI finishes all jobs, download and flash the image.
```

File: source/user-guide/lmp-device-auto-register/lmp-device-auto-register.rst [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+Testing Auto Register
+^^^^^^^^^^^^^^^^^^^^^
+
+After boot the new image, if your device is connected to the internet, the device
```
Comment (vanmaegima) [inline source/user-guide/lmp-device-auto-register/lmp-device-auto-register.rst:138]:
```suggestion
After booting the new image, if your device is connected to the internet, the device 
```

File: source/user-guide/lmp-device-auto-register/lmp-device-auto-register.rst [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+The latest **Target** named ``platform-devel`` should be the CI job you just created.
+
+When FoundriesFactory CI finishes all jobs, download and flashes the image.
+
+Testing Auto Register
+^^^^^^^^^^^^^^^^^^^^^
+
+After boot the new image, if your device is connected to the internet, the device 
+will automatically registerer and should be visible by navigating to the web
```
Comment (vanmaegima) [inline source/user-guide/lmp-device-auto-register/lmp-device-auto-register.rst:139]:
```suggestion
will automatically register to your factory and should be visible by navigating to the web 
```

File: source/reference-manual/security/factory-registration-ref.rst [hunk 5]
```diff
@@ -3,9 +3,9 @@
 Manufacturing Process for Device Registration
 =============================================
 
-lmp-device-register works great when run manually and can be configured
+lmp-device-auto-register works great when run manually and can be configured
```
Comment (angolini) [inline source/reference-manual/security/factory-registration-ref.rst:6]: I think it's confusing to say that a "auto" tool works great when run manually. Maybe reword a little bit to adjust it

File: source/user-guide/lmp-device-auto-register/lmp-device-auto-register.rst [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+
+Prerequisites
+-------------
+
+To follow this section, it is important to have:
+
+- Completed the :ref:`sec-learn` until the :ref:`gs-flash-device` section.
+
+Instructions
+------------
```
Comment (angolini) [inline source/user-guide/lmp-device-auto-register/lmp-device-auto-register.rst:27]: The same as the other PR. I think a reader coming to a UG wants more direct information, sometimes punctual information on how to perform one subtask.

I think we can drop the "Instructions" without changing the content.

File: source/user-guide/lmp-device-auto-register/lmp-device-auto-register.rst (hunk 1 above)
Comment (doanac) [reply source/user-guide/lmp-device-auto-register/lmp-device-auto-register.rst:14]: i would say to not use an API *key* for production. The recipe can still work as-is with the factory-registration-reference server.

File: source/user-guide/lmp-device-auto-register/lmp-device-auto-register.rst [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+
+Select the :guilabel:`device:create` token and 
+select your **Factory**. You can later revoke this access and set up a new 
+token once you are familiar with the :ref:`ref-api-access`.
+
+.. figure:: /_static/userguide/lmp-device-auto-register/lmp-device-auto-register-token.png
+   :width: 500
+   :align: center
+
+   Token for source code access
```
Comment (doanac) [inline source/user-guide/lmp-device-auto-register/lmp-device-auto-register.rst:52]:
```suggestion
   Token for device creation access
```

File: source/user-guide/lmp-device-auto-register/lmp-device-auto-register.rst [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+To follow this section, it is important to have:
+
+- Completed the :ref:`sec-learn` until the :ref:`gs-flash-device` section.
+
+Instructions
+------------
+
+Creating Token
+^^^^^^^^^^^^^^
+In the right top corner, click on the avatar and select :guilabel:`Settings` in the
```
Comment (angolini) [inline source/user-guide/lmp-device-auto-register/lmp-device-auto-register.rst:31]: I understand it's from the Foundries.io Dashboard, is that right?

File: source/user-guide/lmp-device-auto-register/lmp-device-auto-register.rst [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+     lmp-device-auto-register.service - Script to auto-register device into Factory
+          Loaded: loaded (/usr/lib/systemd/system/lmp-device-auto-register.service; enabled; vendor preset: enabled)
+          Active: active (exited) since Sun 2021-09-12 17:34:06 UTC; 5min ago
+         Process: 774 ExecStart=/usr/bin/lmp-device-auto-register (code=exited, status=0/SUCCESS)
+        Main PID: 774 (code=exited, status=0/SUCCESS)
+
+Summary
+^^^^^^^
+
+This user guide gives you an example of how to enable the LmP Device Auto Registered.
```
Comment (angolini) [inline source/user-guide/lmp-device-auto-register/lmp-device-auto-register.rst:181]: The User Guide is the collection of several chapters, this page is one chapter.

I would drop the summary for the user guide entries, as I think the purpose of UG is to be more direct and punctual.

**Review by angolini — COMMENTED** [review]
Great PR! I have left only minor comments

File: source/reference-manual/security/factory-registration-ref.rst (hunk 5 above)
Comment (munoz0raul) [reply source/reference-manual/security/factory-registration-ref.rst:6]: That sentence was there already. I just update the lmp-device-register to the correct name witch is lmp-device-auto-register. 
I don't want to update this sentence in this PR because it is not the focus here. If it is not good enough we can create a new PR to update the /reference-manual/security/factory-registration-ref.rst

File: source/user-guide/lmp-device-auto-register/lmp-device-auto-register.rst (hunk 8 above)
Comment (munoz0raul) [reply source/user-guide/lmp-device-auto-register/lmp-device-auto-register.rst:31]: Made it much more clear now.

**Comment by doanac** [comment]
Docs for 12da2877b5e710bda87da288bc5b5684b11031d8 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1114/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for 73108aef0ee00a03284a0009b2ec400159e3213e are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1116/docs/artifacts/html/index.html

File: source/user-guide/lmp-device-auto-register/lmp-device-auto-register.rst (hunk 9 above)
Comment (munoz0raul) [reply source/user-guide/lmp-device-auto-register/lmp-device-auto-register.rst]: I accepted your suggestions on removing the prerequisites, summary, and instructions.

**Review by angolini — APPROVED** [review]

**Comment by doanac** [comment]
Docs for bd718ba9b50ac393653339e3b4a107bf64927064 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1118/docs/artifacts/html/index.html

**Review by vanmaegima — APPROVED** [review]
